conn.go

```diff
@@ -941,13 +947,18 @@ func (c *Conn) close(byUser bool) error {
 	if byUser {
 		c.connectionClosedByUser = true
 	}
+	isClosed := c.isConnectionClosed()
 	c.closed.Close()
 	c.closeLock.Unlock()
 
 	if closedByUser {
 		return ErrConnClosed
 	}
 
+	if isClosed {
+		return nil
```

Comment:
`crypto/tls.Conn` (based on `net.TCPConn`) seems to return `use of closed network connection` error if closed twice.

Reply:
Never mind.
In the current implementation, first `Close()` after timeout returns `nil` by this condition and second `Close()` returns `ErrConnClosed` by the above condition. It matches `crypto/tls.Conn`'s behavior.